plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1139,7 +1139,12 @@ public void dropPartition(String databaseName, String tableName, List<String> pa
                     .stopOnIllegalExceptions()
                     .run("dropPartition", stats.getDropPartition().wrap(() -> {
                         try (ThriftMetastoreClient client = createMetastoreClient()) {
+                            Partition partition = client.getPartition(databaseName, tableName, parts);
                             client.dropPartition(databaseName, tableName, parts, deleteData);
+                            String partitionLocation = partition.getSd().getLocation();
```

Comment:
Should we wrap the file deletion in a separate try/catch?

I'm just thinking, if the partition is dropped from the metastore but the file system doesn't allow us to delete the files for some reason, the query should still succeed right?

Reply:
I feel users should disable it in such cases. This feature can be controlled by config property, so exception looks better in my opinion.

Reply:
I think it depends on what the behavior is for a following `select` statement. If I run a drop partition and it fails, the partition should still show up in the next query. If it doesn't show up, the drop should succeed. Does that make sense?

Reply:
I was missing that subsequent `deleteDirRecursive` already has try-catch block in the method. Do you think it still requires additional change?

Reply:
You're totally right. This is fine